migrator/migration-steps.js

```diff
@@ -242,6 +241,95 @@ async function updateUnbrandedLayouts (dir) {
   return results.flat()
 }
 
+async function upgradeIfPossible (filePath, matchFound) {
+  if (matchFound) {
+    return true
+  } else {
+    const reporter = await addReporter(`Update ${filePath}`)
+    const fullPath = path.join(projectDir, filePath)
+    const filename = fullPath.split(path.sep).pop()
+    if (filename === 'application.js') {
+      await upgradeApplicationJs(fullPath, reporter)
+      return true
+    }
+    await reporter(false)
+    return false
+  }
+}
+
+async function upgradeApplicationJs (fullPath, reporter) {
+  const textToReplace = [
+    {
+      originalText: '/* global $ */',
+      replacementText:
+        `//
+// For guidance on how to add JavaScript see:
+// https://prototype-kit.service.gov.uk/docs/adding-css-javascript-and-images
+//`
+    },
+    {
+      originalText:
+        `// Warn about using the kit in production
+if (window.console && window.console.info) {
+window.console.info('GOV.UK Prototype Kit - do not use for production')
+}`
+    },
+    {
+      originalText: '$(document).ready(function () {',
+      replacementText: 'window.GOVUKPrototypeKit.documentReady(() => {'
+    },
+    {
+      originalText: 'window.GOVUKFrontend.initAll()',
+      replacementText: '  // Add JavaScript here'
+    }
+  ]
+  const fileBuffer = await fsp.readFile(fullPath)
+  const fileContent = fileBuffer.toString()
+  const fileLines = fileContent.split('\n')
+  const newContentLines = []
+  let lastPos = 0
+  textToReplace.forEach(({ originalText, replacementText }) => {
+    const originalLines = originalText.split('\n')
+    // Find the original text in the file
+    const startPos = fileLines.findIndex((line) => line.trim() === originalLines[0].trim())
+    if (startPos === -1) {
+      return
+    }
+    lastPos = lastPos + originalLines.length
+    while (lastPos < startPos) {
+      newContentLines.push(fileLines[lastPos++])
+    }
+
+    if (replacementText) {
+      // Make sure all original lines exist in sequence within the file before replacing them
+      const canReplace = originalLines.every((originalLine, index) => {
+        return originalLine.trim() === fileLines[startPos + index].trim()
+      })
+      if (canReplace) {
+        const replacementLines = replacementText.split('\n')
+        for (let i = 0; i < replacementLines.length; i++) {
+          newContentLines.push(replacementLines[i])
+        }
+      }
+    }
+  })
+  if (newContentLines.length) {
+    lastPos++
+    while (lastPos < fileLines.length) {
+      newContentLines.push(fileLines[lastPos++])
+    }
+    let content = newContentLines.join('\n')
+
+    // Put back the global line to satisfy linter if jQuery is still necessary
+    if (content.includes('$(')) {
```

Comment:
It might be worth catering for `$.` as well.

Reply:
we've added this to the conditional statement